controller/src/main/java/io/pravega/controller/server/eventProcessor/requesthandlers/DeleteStreamTask.java

```diff
@@ -73,7 +73,9 @@ public DeleteStreamTask(final StreamMetadataTasks streamMetadataTasks,
         return streamMetadataStore.getSegmentCount(scope, stream, context, executor)
                 .thenComposeAsync(count ->
                         streamMetadataTasks.notifyDeleteSegments(scope, stream, count)
-                                .thenComposeAsync(x -> streamMetadataStore.deleteStream(scope, stream, context,
+                            .thenComposeAsync(x -> streamMetadataStore.removeStreamFromAutoRetention(scope, stream, context,
+                                executor), executor)
+                            .thenComposeAsync(x -> streamMetadataStore.deleteStream(scope, stream, context,
```

Comment:
Should we invert the order here to delete then remove? I'm thinking about an anomaly in which we remove it but fail to delete, so the stream will keep going without retention, although eventually the delete task should complete.

Reply:
Yes, delete task should eventually complete. 
Also, its a sealed stream, so all we are doing is removing retention before deleting the stream. So its not like there is additional data being added to the stream.